### ISSUE TITLE
Expose information about the `Voter` to outside

### ISSUE THREAD

**Issue by rphmeier** [paritytech/finality-grandpa#55]
Right now, the only interaction that `Voter` has with the outside is informing the `Environment` when it starts a new round, completes one, finalizes a block, or detects an equivocation.

We should add a new type

```rust
#[derive(Clone)]
struct VoterState {
    inner: Arc<parking_lot::Mutex<Inner>>,
}

impl VoterState {
    // information like proposer, round number, what we've done in the round, etc.
    pub fn active_rounds(&self) -> Vec<RoundData>;
}

impl Voter {
    pub fn voter_state(&self) -> VoterState;
}
```

The `inner` member of `VoterState` should be updated by the voter whenever a round is added or dropped.

**Comment by rphmeier** [comment]
Also, we want to expose an API for passing a callback along with a global message that is called with various return codes based on how useful or valid the message was.

**Comment by andresilva** [comment]
Fixed in #114.
